### ISSUE TITLE
[Device Support Request] Sonoff SWV-BSP Smart Water Valve

### ISSUE THREAD

**Issue by andy-81** [zigpy/zha-device-handlers#3298]
### Problem description

Currently only opening and closing the valve is possible however the Sonoff hub and Zigbee2MQTT support pulling in if there is an issue with the valve and the flow data from the valve which would be fantastic to be able to use in ZHA.

### Solution description

It would be great if a custom quirk could be created however I have struggled to follow along with the details of how to create the custom device handlers. I was hoping the work from Z2M may help point to how to pull the information but the information does not look similar to the Zwave standard where you require the address details which has further complicated me understanding how to write the custom driver. Any help in someone creating this custom handler would be appreciated.

### Screenshots/Video

<details><summary>Screenshots/Video</summary>

https://youtu.be/ofzIoxliNJo?si=9JpD2oPFi9j76muC

this video shows the details of what is shown in ZHA vs what is shown in Zigbee2MQTT

</details>

### Device signature

<details><summary>Device signature</summary>

```json
[Paste the device signature here]
```

</details>

### Diagnostic information

<details><summary>Diagnostic information</summary>

```json
[Paste the diagnostic information here]
```

</details>

### Logs

<details><summary>Logs</summary>

```python
[Paste the logs here]
```

</details>

### Custom quirk

<details><summary>Custom quirk</summary>

```python
[Paste your custom quirk here]
```

</details>

**Comment by AKTheKnight** [comment]
Just got one of these and can provide some of the info missing above:

Signature:
```
{
  "node_descriptor": {
    "logical_type": 2,
    "complex_descriptor_available": 0,
    "user_descriptor_available": 0,
    "reserved": 0,
    "aps_flags": 0,
    "frequency_band": 8,
    "mac_capability_flags": 128,
    "manufacturer_code": 4742,
    "maximum_buffer_size": 82,
    "maximum_incoming_transfer_size": 255,
    "server_mask": 11264,
    "maximum_outgoing_transfer_size": 255,
    "descriptor_capability_field": 0
  },
  "endpoints": {
    "1": {
      "profile_id": "0x0104",
      "device_type": "0x0002",
      "input_clusters": [
        "0x0000",
        "0x0001",
        "0x0003",
        "0x0006",
        "0x0020",
        "0x0404",
        "0x0b05",
        "0xfc11",
        "0xfc57"
      ],
      "output_clusters": [
        "0x000a",
        "0x0019"
      ]
    }
  },
  "manufacturer": "SONOFF",
  "model": "SWV",
  "class": "zigpy.device.Device"
}
```

Diagnostic info attached.
[zha-a1c1fa629b838dd9fa7a401ddd69c026-SONOFF SWV-a6d24191ccfc18b346afd34da19b81e7 (1).json](https://github.com/user-attachments/files/16591561/zha-a1c1fa629b838dd9fa7a401ddd69c026-SONOFF.SWV-a6d24191ccfc18b346afd34da19b81e7.1.json)


Let me know if there's more info I can provide

**Comment by brodock** [comment]
Cluster `0xfc11` seems to be known by zigbee2mqtt project: https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/devices/sonoff.ts#L52-L71

**Comment by brodock** [comment]
schedule seems to be handled as part of the custom cluster above in the following attribute `0x5008`: 

the logic to serialize/de-serialize its content can be seen here: 
https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/devices/sonoff.ts#L297-L370

**Comment by andy-81** [comment]
I did find reference in the zigbee2mqtt handler about the SWV valve.

It does mention in the list about the water flow and the status which shows if it is leaking, no pressure... 

It can be found here:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/devices/sonoff.ts#L1104-L1138

**Comment by jamesshannon** [comment]
[This is the CL](https://github.com/Koenkk/zigbee-herdsman-converters/pull/7132/files) that added much of the code. Notice that they also had to edit the `fromZigbee.ts` file to add the `msFlowMeasurement`... uh... "thing" (I don't know much about zigbee). Not sure if equivalent work would need to be done here.

**Comment by soyxan** [comment]
I am also interested in this device support. I own the device and can provide any information or perform any test if needed. Thanks!

**Comment by benbancroft** [comment]
Hi all,

I've just created PR https://github.com/zigpy/zha/pull/187 for zha, which implements at least the flow rate sensor part of this request. I plan to look at implementing a device handler for `0xfc11` cluster features (i.e. issue status) next if I have some time in coming days/weeks, unless somebody else has already made some progress here?

**Comment by benbancroft** [comment]
I've just pushed up https://github.com/zigpy/zha-device-handlers/pull/3340 and https://github.com/zigpy/zha/pull/189, which implements device quirk and Home Assistant sensors for valve leak status and water supply status. Other than some device weirdness around having to ensure manufacturer_specific is not set on attribute read, this wasn't too bad to add in the end.

This is working nicely for my irrigation setup so far:
![Screenshot 2024-08-31 201915](https://github.com/user-attachments/assets/3b1e16c0-4375-47bb-bc6f-66d2fda52118)

That should hopefully be everything that was initially requested for this request. There are however other features supported in Zigbee2MQTT (scheduling, shutoff timer etc) that probably could be added as a future enhance, which again I might look at if I get some time.

**Comment by soyxan** [comment]
Thanks a lot, that is a good improvement. Now a newbie question, when will this quirck be available in Home Assistant? Can I manually upgrade ZHA or do I have to wait for a new HA release?

On the other hand, regarding the other features available in Zigbee2MQTT like scheduling (timed and quantitative), I think that is quite important to have them available. Rely in HA for scheduling the irrigation maybe is not the best option as any failure will affect our plants wellbeing :) Been able to define and save the schedules in the SWV device will keep the irrigation working even if HA or the Zigbee network is completely dead. I do not have the skills to do it by my own, but I am available for testing as I own the SWV.

**Comment by aaronegger** [comment]
Is there any chance to get the amount of water (total liters)? That would be super usefull :D

**Comment by jamesshannon** [comment]
My two cents regarding scheduling, specifically turn-on scheduling. The device will be fairly simple (e.g., time of day, though it might have another way to schedule). It definitely won't have sun-based scheduling, forecast-based scheduling, etc. So it's kinda a weird User Experience if it supports some on-device scheduling but not others.

Now... if you're talking about "time / flow *limits*" then yes, I agree that's super important. When I turn on the faucet (via HA) then I want to be able to send a "turn off in X" parameter as part of the "on" message. That way if something happens to the network then the device still turns off. I'm a little concerned about my plants, but very concerned about my water running for days. The issue could be something as simple as me rebooting HA at the time that my "turn off" automation is supposed to run.

**Comment by TeDeVPrime** [comment]
nice! but how do we get updates for the ZHA in Hass? i don't see these options in my list

**Comment by TeDeVPrime** [comment]
guys anyone who can tell us how to get the latest updates?

**Comment by brodock** [comment]
it seems the related PRs haven't been merged yet

**Comment by andy-81** [comment]
Any idea when this pr will be merged?

Would be great to gain access to the features @benbancroft added already. 😊

**Comment by TeDeVPrime** [comment]
yeah i have been checking with every update but no luck.
i already had a lot of issues with my water valve left open. i really need to be able to use the on-device timer.

i don't know how ZHA works with HASS, but shouldn't it be able to update on it's own?

**Comment by TeDeVPrime** [comment]
guys the new version from HASS came up but still my Sonoff Smart Water Valve just shows on / off switch. nothing more.

no i need to do something?

**Comment by jamesshannon** [comment]
The pull requests haven't been approved. After they're approved and merged into the production branch then the updated version of zigpy will need to be included in ZHA, and then ZHA will need to be included in HA. In theory, all these things could happen quickly, but they require approval and merging.

**Comment by TeDeVPrime** [comment]
i find it crazy that a pull request would need months.
ZHA should be able to be updated on it's own and not as part of HA. 
what if we need a hotfix?

we have had two HA releases since this was fixed. now i have to wait for November (if i am lucky) in order to be able to control my watering safely.

two times it has failed me until now and the water left running.

**Comment by jamesshannon** [comment]
I can't speak to the approval process. It might have just gotten forgotten about.

I don't know why you're having so many problems. If you have an HA automation set up that turns it on, then waits, then turns it off, you should only get failures if:
a) The zigbee connection is really flaky. But if that's the case then you probably have failed turn-ons, too?
b) HA dies (or gets restarted, etc) during the on time and then the automation doesn't finish to turn it off.

For:
a)  I believe that Zigbee has a bi-directional mechanism and includes confirmation, right? In which case you should be able to build a check-and-retry into your automation. Ie, "repeat: turn off. wait 10 seconds. if it's off then exit" 

b) You could build a "failsafe off" automation to run, e.g., every 15 minutes and turn the water off. The more frequently it runs and the longer your watering period the more likely you are to have issues with this (ie, you'll turn off while it should still be one). If that's a problem, you could add a datetime helper or something as a second check.

**Comment by TeDeVPrime** [comment]
i had two failures 

1 was because my HASS crashed..... lucky eh?
and 1 was because the device became "unavailable" for whatever reason.

i also have a friend who had this issue with bad signal

so i don't want to be dependent on critical things like that to HASS. since the device supports natively the solution to my problem.... and since it is fixed... i want that hotfix. because to me... this is a hotfix.

so having all these steps for "approval" for something that i should be able to access as beta or early-release..... it's annoying. especially when HASS has update for every single thing.... except ZHA.

**Comment by soyxan** [comment]
First of all, I totally agree with the comments regarding the long waiting till we get the update, but, in addition to that and AFAIK, the PR https://github.com/zigpy/zha-device-handlers/pull/3340 only covers basic feedback sensors from the device (flow measurement and leakage status) but no specific commands like "turn on with shutdown timer" as some of you are expecting. Please correct me if I am wrong.

**Comment by TeDeVPrime** [comment]
you are correct. i am hoping that someone will soon figure it out. from what i have read, sonoff made it in a way that when you send the turn on command... you can also pass parameters for water flow and time.

the sensors that "exists" at the moment will give me an extra way of checking the system on event changes. 
it will also give me notifications as to whether i have running water or not at my house (it's a weekend house, so i am not there every single day).

i like the simplicity of ZHA but man.... Z2M had these features since day one. 

i have never gotten to the technical details between these two (ZHA and Z2M). i just found the ZHA easier and faster to use. 
Hopefully we will see updates on this device soon

**Comment by fgsch** [comment]
Can we please move the discussion elsewhere, e.g., GH discussions?
If you want to be notified of updates just subscribe to the PR.  Protesting here doesn't help.

**Comment by marcforn** [comment]
I'm also waiting for this to be merged

**Comment by micz** [comment]
I created PR #3930 that adds these features:

- Water Leakage binary sensor
- Water Shortage binary sensor
- `on_time` attribute, representing the number of seconds after opening before the valve automatically closes.
- A custom behaviour for the On command:
   1. If `on_time == 0` it's a normal On command (`0x01`)
   2. If `on_time != 0` it sends a on_with_timed_off (`0x42`), with the `on_time` numbers of seconds sets in the attribute

In my environment everything works, could someone help testing it?
Thank you!

**Comment by Stothed775** [comment]
Has an “open for” function already been implemented? So that the valve closes again automatically after a set time?

In my opinion, this is a very important safety feature

**Comment by micz** [comment]
With PR #3930, I tried to implement that. It has been labeled "waiting for reviewer response".

I'm setting up the system for the spring and plan to use it as a custom quirk.

**Comment by PrimesoftCopilot** [comment]
it is good seeing someone trying to make this device work properly.
ZHA "zigpy" has been a mess around this device since day one.
Z2M had full support since day one.
6-7 months after product release and we are still waiting for "approvals".

i must be missing a lot of knowledge or information because the fact that we have to wait for someone's "approval - review"... and then wait for the next version of HASS that will include the changes.... it is simply nuts.

Micz thank you for your effort. I don't know if we are going to magically see the changes when they happen or we have to reconfigure the device but at least you made a progress.

**Comment by etienne678** [comment]
i took the quirk code from PR #3930 and edited it to hardcode a maximum of 20 minutes valve open time and use it always when the device is turned on, so that even if i forget or connection is lost it turns off after 20 mins. I have this running in my garden with 4 of those devices. it works well for me like this, but the on_time should be a setting in the UI. unfortunately setting the kwargs["on_time"] through the UI did not work in my case.

https://gist.github.com/etienne678/e2670600872c5b4021f9ec390edae661

**Comment by micz** [comment]
@etienne678 which firmware version do you have on the SWVs?

**Comment by etienne678** [comment]
@micz  
in HA it says:
Installed version
0x00001003
Latest version
0x00001003

i updated it through the ewelink app last week before i switched them over to HA / ZHA. 
if above is not enough, please let me know how to look it up i can post it here.

**Comment by micz** [comment]
I have 0x00001004, maybe that's why it's not working. I never tried it with 1003.

**Comment by etienne678** [comment]
<img width="556" alt="Image" src="https://github.com/user-attachments/assets/c9002d63-8cde-4683-ae6e-dc2d5037f102" />

no newer version available, at least in HA.

**Comment by micz** [comment]
Before starting the work on this quirk I set up a test network with a spare coordinator using zigbee2mqtt to see how the device works there. So the device was updated to 1004.
Then I setup a test network with ZHA dev environment to work on the quirk, but the device was already using 1004.

**Comment by Antriel** [comment]
What needs to happen to get this done, and get the new firmware to be available via ZHA?

**Comment by gianlucaromito** [comment]
I'm using the quirk too and it works good with 1004. Before (1002) it did not work setting the auto close time from the UI.
Waiting to see this implemented in ZHA.

**Comment by austwhite** [comment]
I am using ZHA and updated to Firmware 1004 now.   The device was fine on 1003, but Itead / SOnOff did add the option to disable/change the auto-shutoff on low water flow.   This was an update on 1004
This option is available in Zigbee2MQTT, but not available in ZHA yet.
I saw about a quirk above, but not 100% sure how to implement it.  
Mainly  I am leaving a comment to watch this :)

**Comment by petrikd** [comment]
Hi All!
Maybe this is not the right thread - But does anyone know why is "Unknown" the SWV firmware in ZHA (for me)?
I'm really interested in this topic - can wait to possibility to set the auto close time :D

**Comment by austwhite** [comment]
@petrikd 
Likely firmware settings in your configuration.yaml file, but this is not the place to discuss it.
This is a request for a modification to a driver (which is ongoing) and not a discussion forum. 
There should be plenty of discussion forums if you do a google search.   That's how I found the correct lines to add to my file   :)

**Comment by TeDeVPrime** [comment]
is there any news for this?
i don't even see an update for this device in HASS... i don't see any other options other than the basic....

**Comment by austwhite** [comment]
@TeDeVPrime 
the update to firmware is not going to bring the extra features to ZHA.   We just have to wait for the update to the quirk / device driver.
If you do want to try firmware I went to this discussion, updated mine fine via ZHA.   Just do not expect to see extra features as you won't get any.   Firmware should be discussed elsewhere though.
https://community.home-assistant.io/t/sonoff-swv-water-valve-auto-turn-off/754319/39?page=2

**Comment by c3poSAW** [comment]
Hi, I'm very new to HA.

I jutst moved nearly all my devices to HA Zigbee. but with this device i strugle. In my old setup i used a tasmota Zigbeebridge and used node Red to get my datas out. But here im Lost. Maybe you can help or explain, how to get now the Daily Volume and the Total.

Tasmota Bridge Log: 
16:56:41.092 MQT: tele/ZigbeePro/SENSOR = {"ZbReceived":{"0xA6E7":{"Device":"0xA6E7","Name":"WasserhanDraußen","Cluster":10,"Read":[7],"ReadNames":{"LocalTime":true},"Endpoint":1,"LinkQuality":72}}} 16:56:41.634 MQT: tele/ZigbeePro/SENSOR = {"ZbReceived":{"0xA6E7":{"Device":"0xA6E7","Name":"WasserhanDraußen","Power":0,"Endpoint":1,"LinkQuality":76}}} 16:56:43.627 MQT: tele/ZigbeePro/SENSOR = {"ZbReceived":{"0xA6E7":{"Device":"0xA6E7","Name":"WasserhanDraußen","BatteryPercentage":50,"Endpoint":1,"LinkQuality":69}}} 16:56:47.592 MQT: tele/ZigbeePro/SENSOR = {"ZbReceived":{"0xA6E7":{"Device":"0xA6E7","Name":"WasserhanDraußen","FC11/5006":62,"FC11/5007":692,"FC11/500D":812138946,"FC11/500E":812139008,"Endpoint":1,"LinkQuality":69}}} 16:56:49.589 MQT: tele/ZigbeePro/SENSOR = {"ZbReceived":{"0xA6E7":{"Device":"0xA6E7","Name":"WasserhanDraußen","FC11/500F":0,"FC11/5010":0,"FC11/500C":1,"Endpoint":1,"LinkQuality":69}}}

Hier der Code aus meinem Node Red:
// Output 1: lastValveOpenDuration (Sekunden) outputs[0] = sensorData["FC11/5006"] !== undefined ? createMsg(sensorData["FC11/5006"], "lastValveOpenDuration", "s") : null; 

// Output 2: irrigationVolume (Liter, Faktor 0.1) outputs[1] = sensorData["FC11/5007"] !== undefined ? createMsg(sensorData["FC11/5007"] * 0.1, "irrigationVolume", "l") : null; 

// Output 3: valveStatus (0 = geschlossen, 1 = offen) outputs[2] = sensorData["FC11/500C"] !== undefined ? createMsg(sensorData["FC11/500C"], "valveStatus") : null; 

// Output 4: irrigationStartTime (Unix-Zeitstempel) outputs[3] = sensorData["FC11/500D"] !== undefined ? createMsg(sensorData["FC11/500D"], "irrigationStartTime") : null; 

// Output 5: irrigationEndTime (Unix-Zeitstempel) outputs[4] = sensorData["FC11/500E"] !== undefined ? createMsg(sensorData["FC11/500E"], "irrigationEndTime") : null; 

// Output 6: dailyIrrigationVolume (Liter) outputs[5] = sensorData["FC11/500F"] !== undefined ? createMsg(sensorData["FC11/500F"], "dailyIrrigationVolume", "l") : null; 

// Output 7: workState (Statuscode) outputs[6] = sensorData["FC11/5010"] !== undefined ? createMsg(sensorData["FC11/5010"], "workState")

vielleicht kann ja jemand damit was anfangen?

**Comment by PaulBol** [comment]
@c3poSAW austwhite already mentioned it above: This is a request for a modification to a driver (which is ongoing) and not a discussion forum.

**Comment by Hedda** [comment]
Also be aware of this related Device Support Request in the zha-device-handlers repo which asks for other attributes to be exposed:

* https://github.com/zigpy/zha-device-handlers/issues/4340

There is still also an open pull request from @micz that will expose Auto Close Timer (if using the latest firmware on the device):

* https://github.com/zigpy/zha-device-handlers/pull/3930

FYI, Sonoff have also released four additional models of they Zigbee Smart Water Valve and all looks to expose the same:

* Sonoff SWV (original model with two different fittings) -> https://www.zigbee2mqtt.io/devices/SWV.html
  * Note that SWV-BSP and SWV-NH is the same product shipping with different fittings for Europe and North America.
* Sonoff SWV-ZFE (Hydro Series with Flow Meter) -> https://www.zigbee2mqtt.io/devices/SWV-ZFE.html
* Sonoff SWV-ZFU (Hydro Series with Flow Meter) -> https://www.zigbee2mqtt.io/devices/SWV-ZFU.html
* Sonoff SWV-ZNE  (Hydro Series without Flow Meter) -> https://www.zigbee2mqtt.io/devices/SWV-ZNE.html
* Sonoff SWV-ZNU (Hydro Series without Flow Meter) -> https://www.zigbee2mqtt.io/devices/SWV-ZNU.html

Update! Itead have now also released an Zigbee dual-channel smart water valve called Sonoff SWV-ZF2 as well:

* Sonoff SWV-ZF2 (dual-channel smart water valve) -> https://www.zigbee2mqtt.io/devices/SWV-ZF2.html
  * Note that SWV-ZF2U and SWV-ZF2E  is the same product shipping with different fittings for North America and Europe.

PS: Slightly off-topic but for reference also be aware of these discussions about adding support for the Valve platform to ZHA, see:

* https://github.com/zigpy/zha/pull/673

and

  * https://github.com/zigpy/zha/issues/528

**Comment by Hedda** [comment]
> be aware of this related Device Support Request in the zha-device-handlers repo which asks for other attributes to be exposed:
> 
> * [[Device Support Request] Sonoff SWV Smart Water Valve - timer function #4340](https://github.com/zigpy/zha-device-handlers/issues/4340)

Be sure to also keep an eye on https://github.com/zigpy/zha-device-handlers/issues/4340 as there are now several new related pull requests linked from that, so check out these too:

* https://github.com/zigpy/zha-device-handlers/pull/4927

and

* https://github.com/zigpy/zha-device-handlers/pull/4993

**Comment by X-Ryl669** [comment]
Notice that for my SWV-ZFE devices, by default:
1. Switching the valve on auto switches off after 10mn (there's a default 10mn timer off in the device)
2. Using OnTime / Offtime (0x4001, 0x4002) attributes of cluster id 0x06 (OnOff) isn't supported by the device, opposite to what the quirks file in #4927 is proposing. The device answer with an Unsupported Attribute error
3. This 10mn timer is a default, but I haven't found a way to disable it. Seems like @ausfas found out that using a "On with Timed off" command (0x42) on cluster id 0x6 works, but in a unusual way: you need to set the "On/Off control" to 0 and set both the "On time" and "Off time"). So switching the valve on with "On" command (0x01) doesn't work as expected (it turns the valve on, but auto turns off for a non-user selected time which breaks irrigation schedules for example). It is **mandatory** to set a timer for using this device and none of the quirks above do that.

Looking at [Zigbee protocol](https://zigbeealliance.org/wp-content/uploads/2019/12/07-5123-06-zigbee-cluster-library-specification.pdf) page 3-56, and opposite to @ausfas is doing [here](https://github.com/zigpy/zha-device-handlers/pull/3930#issuecomment-4250761470), the `ontime` and `offtime` are supposed to be in 1/10 of seconds (so a value of 600 is 60s not 10mn).

**Comment by ausfas** [comment]
I have been testing the SONOFF SWV on ZHA and found that the valve supports local timed operation through the standard On/Off cluster command `OnWithTimedOff`.

This is a Zigbee cluster command, not an attribute write.

The command is:

* Endpoint: `1`
* Cluster: `0x0006` / OnOff
* Command: `0x42` / OnWithTimedOff
* Parameters:

  * `on_off_control`: `0`
  * `on_time`: duration in seconds
  * `off_wait_time`: `1`

The important part is that once this command is accepted by the device, the countdown is handled locally by the valve. Home Assistant does not need to remain online for the valve to close at the end of the timer.

For example, this works using Home Assistant’s native ZHA service:

```yaml
action: zha.issue_zigbee_cluster_command
data:
  ieee: 00:11:22:33:44:55:66:77
  endpoint_id: 1
  cluster_type: in
  command_type: server
  cluster_id: 0x0006
  command: 0x42
  params:
    on_off_control: 0
    on_time: 36
    off_wait_time: 1
```

The same command can also be sent using ZHA Toolkit, but the field names are different. With `zha_toolkit.zcl_cmd`, use `endpoint`, `cluster`, and `cmd` rather than `endpoint_id`, `cluster_id`, and `command`:

```yaml
action: zha_toolkit.zcl_cmd
data:
  ieee: 00:11:22:33:44:55:66:77
  endpoint: 1
  cluster: 0x0006
  cmd: 0x42
  args:
    - 0
    - 36
    - 1
```

For a 3-hour timer, the second argument / `on_time` value is `10800` seconds:

```yaml
action: zha_toolkit.zcl_cmd
data:
  ieee: 00:11:22:33:44:55:66:77
  endpoint: 1
  cluster: 0x0006
  cmd: 0x42
  args:
    - 0
    - 10800
    - 1
```

Plain OFF can be sent with:

```yaml
action: zha_toolkit.zcl_cmd
data:
  ieee: 00:11:22:33:44:55:66:77
  endpoint: 1
  cluster: 0x0006
  cmd: 0x00
```

In my Home Assistant automation I calculate the selected duration into seconds and pass that value as `on_time`:

```yaml
params:
  on_off_control: 0
  on_time: "{{ secs | int }}"
  off_wait_time: 1
```

This gives me two layers of safety:

1. The SWV itself closes locally using `OnWithTimedOff`, even if Home Assistant restarts or crashes.
2. Home Assistant also runs a mirror timer for UI/progress display and sends an additional OFF command at the end as a backup.

So the main discovery is that the SWV can be timed locally with `0x0006 / 0x42`, using seconds as the timer value.

**Comment by X-Ryl669** [comment]
In the protocol, they say it should be 1/10th of a second (so 600 is 60s = 1mn). I've tried it with the value set to 60 (so 6s if following the standard, 60s if following your description) and the valve ran for 10mn (so it's completely not standard). I've then tried for 120s  and it run for 10mn as well. So something is fishy here.

**Comment by starkillerOG** [comment]
I just tried: https://github.com/zigpy/zha-device-handlers/pull/5114 and it works greath for the SWV-ZFE and also allows control over the manual irrigation duration and volume. It looks like that quirk was made by a Sonoff employee.
